### PR TITLE
Add `c:indestructible` and `c:shulker_boxes` tags

### LINKED ISSUE #894: Convention tag requests
> [!NOTE]
> Any requests for a new tag under the `c:` namespace should also be mirrored to the https://github.com/FabricMC/fabric repository.
> 
> No change to the tags will be entertained that is not brought up for discussion on Fabric's side. 
> 
> Neither will a PR to these tags be merged unless both NeoForge and Fabric have approved the PR with the same tag(s), the same contents, and the same definition of what the tag(s) contains.
> 
> ~ @sciwhiz12

---

Why is there no shulker box tag? 😭 @TelepathicGrunt

### LINKED ISSUE #941: Add c:indestructible tag
This Issues asks to add the c:indestructible tag for items.

This tag is there to represent items which shall not be destroyed.

I don't know if c:indestructible is the correct naming. But I don't have a better idea.

I have not added a block tag. As I would be unsure if bedrock would go into it? For some mods bedrock may be considered as destructible.

This tag does have no gameplay altering effect. That means that items are still destructible my vanilla methods. This will need to be prevented by the mod itself. The tag is there to tell other mods that something should not be destroyed.  

This issue is there to get feedback from both mod loaders for this.
https://github.com/FabricMC/fabric/pull/3758

### PR DESCRIPTION
**Pull request by Spinoscythe**
Fabric PR for `c:indestructible`: https://github.com/FabricMC/fabric/pull/3758
`c:shulker_boxes` already in Fabric

Fixes #941
Fixes #894

### CODE REVIEW

File: src/generated/resources/reports/registry_order.json [hunk 1]
```diff
@@ -2,6 +2,7 @@
   "order": [
     "minecraft:attribute",
     "minecraft:data_component_type",
+    "minecraft:armor_material",
```
Comment (marchermans) [inline src/generated/resources/reports/registry_order.json]: Why these unneeded changes?

**Review by marchermans — CHANGES_REQUESTED** [review]

File: src/generated/resources/reports/registry_order.json (hunk 1 above)
Comment (Spinoscythe) [reply src/generated/resources/reports/registry_order.json]: it keeps generating them when I run data for some reason

**Review by marchermans — APPROVED** [review]

**Review by Matyrobbrt — CHANGES_REQUESTED** [review]
Was this PR made alongside a Fabric issue or PR? We are not going to merge it until then, and until that PR is approved.

**Comment by Spinoscythe** [comment]
@Matyrobbrt yes there is: https://github.com/FabricMC/fabric/pull/3758

**Comment by TelepathicGrunt** [comment]
@Spinoscythe Could you remove the indestructable tag from this pr? We can merge in the shulker box work. The indestructable tag may need more discussion about it separately

**Comment by Spinoscythe** [comment]
will do when I come back

**Pull request closed without merging**